File: Webshop/Webshop.Api/Entities/Order.cs
using System.ComponentModel.DataAnnotations.Schema;
using Webshop.Api.Interfaces;

namespace Webshop.Api.Entities;

public class Order : IUniqueEntity<Guid>
{
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid Id { get; set; }
    public Guid AddressId { get; set; }
    public Guid UserId { get; set; }
    public OrderStatus Status { get; set; }
    public DateTime CreatedAt { get; set; }
    public Address? Address { get; set; }
    public User? User { get; set; }
    public List<OrderProduct> Products { get; set; } = new();
}

public enum OrderStatus
{
    Open,
    Confirmed,
    Processing,
    Completed,
    Canceled
}

File: Webshop/Webshop.Client/src/app/interfaces/responses/Order.ts
import {Address} from "../Address";
import {OrderProduct} from "./OrderProduct";

export enum OrderStatus {
  Open,
  Confirmed,
  Processing,
  Completed,
  Cancelled,
}

export interface Order {
  id: string;
  addressId: string;
  userId: string;
  status: OrderStatus;
  address: Address;
  products: OrderProduct[];
  totalPrice: number;
  createdAt: Date;
}
